src/main/java/seedu/address/ui/WebViewManager.java

```diff
@@ -0,0 +1,149 @@
+package seedu.address.ui;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.application.Platform;
+import javafx.beans.InvalidationListener;
+import javafx.beans.value.ChangeListener;
+import javafx.concurrent.Worker;
+import javafx.scene.Node;
+import javafx.scene.layout.Pane;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import seedu.address.commons.util.StringUtil;
+
+//@@author qiu-siqi
+/**
+ * Exposes required functionalities of {@link WebView}.
+ *
+ * There can only be one instance of {@code WebViewManager}, allowing the internal limiting
+ * of number of WebViews used.
+ */
+public class WebViewManager {
+
+    private static WebViewManager webViewManager;
+    private WebView browser;
+    private WebEngine engine;
+    private List<InvalidationListener> invalidationListeners;
+    private List<ChangeListener> changeListeners;
```

Comment:
Specify the types of the `ChangeListener`s.
`List<ChangeListener<? super Number>>` and `List<ChangeListener<? super Worker.State>>`